From bdfd2f58fdec502ca5f557079f3c81bfd69f801c Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 19 Nov 2010 03:45:49 +0000
Subject: [PATCH] x25: Disable auto-loading as mitigation against local exploits

Recent review has revealed several bugs in obscure protocol
implementations that can be exploited by local users for denial of
service or privilege escalation.  We can mitigate the effect of any
remaining vulnerabilities in such protocols by preventing unprivileged
users from loading the modules, so that they are only exploitable on
systems where the administrator has chosen to load the protocol.

The 'x25' protocol has been in 'alpha' state for 10 years, and there
appear to be no published applications for Linux that use it.
Therefore disable auto-loading.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 net/x25/af_x25.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/x25/af_x25.c b/net/x25/af_x25.c
index d006816..e0da374 100644
--- a/net/x25/af_x25.c
+++ b/net/x25/af_x25.c
@@ -1743,4 +1743,4 @@ module_exit(x25_exit);
 MODULE_AUTHOR("Jonathan Naylor <g4klx@g4klx.demon.co.uk>");
 MODULE_DESCRIPTION("The X.25 Packet Layer network layer protocol");
 MODULE_LICENSE("GPL");
-MODULE_ALIAS_NETPROTO(PF_X25);
+/* MODULE_ALIAS_NETPROTO(PF_X25); */
-- 
1.7.2.3

